src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Layout/MainLayout.razor

```diff
@@ -15,3 +15,11 @@
         </article>
     </main>
 </div>
+@*#if (UseServer || UseWebAssembly) -->
+
+<div id="blazor-error-ui">
```

Comment:
To stay consistent with the existing Blazor templates, I prefer for the Blazor error UI to remain in the top-level component. I don't really see a good justification for moving it.

Reply:
My justification is:

 * Conceptually it's right for this to be an aspect of layout. People could choose to put this content anywhere in their layout. We shouldn't be so fixated on the default purple-gradient-with-sidebar site. Depending on HTML structure, you may really need this content to be in here.
   * The only reason we didn't do this before was that prerendering wasn't on by default for WebAssembly. Now this component will always be prerendered so there's no problem with the error UI going here.
 * It removes a concern from the top-level HTML that feels unrelated to top-level HTML. It's always been strange to see the phrase "An error has occurred" when you first open the top-level HTML. Anyone would think the site actually displays that phrase by default - the only thing that tells it not to is elsewhere in CSS.

... and I don't think retaining patterns from the existing templates is a goal compared with making the best templates we can for the long-term future. But again, let's discuss!

Reply:
> The only reason we didn't do this before was that prerendering wasn't on by default for WebAssembly. Now this component will always be prerendered so there's no problem with the error UI going here.

Does this mean that if someone decides to turn off prerendering (for example, to avoid having to adjust components that weren't designed to run from the server) then the error UI will break?

Reply:
Unlikely. The error UI is only used for errors during interactive rendering, which can only happen once the app has already started rendering interactively. It may be possible in an edge case to cause an error so early that the layout hasn't yet rendered, but given that layouts render before pages it's really unlikely. And again, would only happen if prerendering is off, and there would still be an error in the dev console.

So altogether I would consider it very reasonable to render the error UI from the layout: it's nicer in the code, and it would be necessary if you intended that UI to fit in your page structure in some way instead of hovering over it. But I do see there's a tradeoff.

Let's discuss later how much of a concern this is to you and what balance we want to strike.

Reply:
We discussed and agreed to leave the error ui here.